File: src/sensor_nivel.py
class SensorNivel:
    def __init__(self, tag: str, valor: float, unidade: str = ""):
        self.tag = tag
        self.valor = valor
        self.unidade = unidade

    def resumo(self) -> str:
        sufixo = f" {self.unidade}" if self.unidade else ""
        return f"{self.tag}: {self.valor:g}{sufixo}"

File: src/sensor_nivel.cpp
#include "sensor_nivel.hpp"

#include <iomanip>
#include <sstream>
#include <utility>

namespace {
std::string formatarValor(double valor) {
    std::ostringstream saida;
    saida << std::setprecision(12) << valor;
    return saida.str();
}
}

SensorNivel::SensorNivel(std::string tagInicial, double valorInicial, std::string unidadeInicial)
    : tag(std::move(tagInicial)), valor(valorInicial), unidade(std::move(unidadeInicial)) {
}

std::string SensorNivel::resumo() const {
    const std::string sufixo = unidade.empty() ? "" : " " + unidade;
    return tag + ": " + formatarValor(valor) + sufixo;
}
